### ISSUE TITLE
GoDaddy Email Sync Breaks on MIME Header Decoding in SuiteCRM 8.4

### ISSUE THREAD

**Issue by DBRenny** [salesagility/SuiteCRM-Core#363]
#### Issue

We are experiencing an issue with the email synchronization functionality in SuiteCRM. Specifically, the system encounters a fatal error during the process of handling MIME header decoding for email subject, to list, from, and other headers. The error message indicates a problem with the foreach() function, suggesting that the variable being iterated upon is not an array or object as expected.
#### Expected Behavior

The CRM should seamlessly decode MIME headers and handle email headers, regardless of their encoding, without encountering fatal errors. The process should iterate over the decoded headers and correctly process the email details into a readable format.

#### Actual Behavior

Instead of successful processing, the system throws a fatal error, particularly pointing to the foreach() line in the handleMimeHeaderDecode method. The error message is: "Invalid argument supplied for foreach()". This issue disrupts the normal flow of email synchronization, preventing emails from being properly imported and read in the CRM.

Relevant logs indicate the error originates from the handleMimeHeaderDecode method, but there's no additional information about the state or format of the data being passed to foreach().

[SuiteCRM_10236 sample log.txt](https://github.com/salesagility/SuiteCRM/files/12920000/SuiteCRM_10236.sample.log.txt)

#### Possible Fix

A potential fix could involve adding robust checks before the foreach() statement to ensure the variable is indeed iterable. If the variable is a JSON string, it might require decoding with json_decode().

Here's a proposed code snippet (GPT helped): It referenced subject, but all header information except time information is not syncing.
```

php
Copy code
// Check if $subjectDecoded is JSON, and decode it if necessary
if (is_string($subjectDecoded)) {
    $decodedJson = json_decode($subjectDecoded);
    if (json_last_error() === JSON_ERROR_NONE) {
        $subjectDecoded = $decodedJson;
    } else {
        // Handle error
    }
}

// Check if $subjectDecoded is iterable
if (is_iterable($subjectDecoded)) {
    // existing foreach loop
} else {
    // Handle error
}

```
#### Steps to Reproduce

1. Set up email synchronization in SuiteCRM.
2. Receive an email with a subject that requires MIME header decoding.
3. Observe the SuiteCRM log or UI for errors during the email import process.
4. Notice the "Invalid argument supplied for foreach()" error pointing to the handleMimeHeaderDecode method.

#### Context

This is our pilot system, but we need to use as many integrated features as possible. I want to get through these roadblocks to start expanding the scope of users.

Outbound email automation still works, and in the meantime my users can use outlook. but this is a major adoptability issue.
Inbound email box view:
![image](https://github.com/salesagility/SuiteCRM/assets/100956107/a1cd68ef-125c-4fb5-bed7-7222c00da8d8)
Email detail level view:
![image](https://github.com/salesagility/SuiteCRM/assets/100956107/bf944e73-09bf-47a9-8e3e-126633247864)

#### Your Environment

* SuiteCRM Version used: 8.4
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version: PHP 8.1.24, MariaDB 11.0.3, Godaddy Office 365 (outlook.office365.com) - utilizing an OATH connection
* Debian GNU/Linux 11 (bullseye)

This is somewhat similar to this issue: [SuiteCRM#9921](https://github.com/salesagility/SuiteCRM-Core/issues/206), but I opened a new one so it could be replicated with my environment.

**Comment by DBRenny** [comment]
I have a fix, but should I leave this open so you can have a PR? Do you want me to place one? I haven't done one of those before on here.

Replacing the function with this resolved my issue:

```
public function handleMimeHeaderDecode($subject)
{
    $GLOBALS['log']->debug('handleMimeHeaderDecode: Starting MIME header decoding for subject: ' . $subject);

    $subjectDecoded = $this->getImap()->MimeHeaderDecode($subject);

    // If $subjectDecoded is a string, log its content to understand why
    if (is_string($subjectDecoded)) {
        $GLOBALS['log']->error('handleMimeHeaderDecode: $subjectDecoded is a string. Content: ' . $subjectDecoded);
        // Instead of returning an error message, we might want to return the original subject
        // or handle this case appropriately based on the actual content of $subjectDecoded.
        return $subject; // returning the original subject for now
    }

    $ret = '';
    if (is_array($subjectDecoded)) { // Checking if it's an array as expected
        $GLOBALS['log']->debug('handleMimeHeaderDecode: $subjectDecoded is an array, proceeding with decoding');
        foreach ($subjectDecoded as $object) {
            if (isset($object->charset) && $object->charset != 'default' && isset($object->text)) {
                $ret .= $this->handleCharsetTranslation($object->text, $object->charset);
            } elseif (isset($object->text)) {
                $ret .= $object->text;
            }
        }
    } else {
        $GLOBALS['log']->error('handleMimeHeaderDecode: $subjectDecoded is not an array');
        // Similar to above, decide how you want to handle this error.
        return $subject; // returning the original subject for now
    }

    $GLOBALS['log']->debug('handleMimeHeaderDecode: Decoding completed, result: ' . $ret);
    return $ret;
}

```

**Comment by clemente-raposo** [comment]
This issue has been fixed on 7.14.4, then ported and released on 8.6.1. Closing

see:
https://github.com/salesagility/SuiteCRM/pull/10323
